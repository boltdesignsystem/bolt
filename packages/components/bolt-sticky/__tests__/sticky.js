import { renderString, stopServer } from '../../../testing/testing-helpers';

describe('<bolt-sticky> Component', () => {
  test('basic usage', async () => {
    afterAll(async () => {
      await stopServer();
    });

    const results = await renderString(
      `
      <div style="height:50vh;background-color:lightgray;"></div>

      {% embed "@bolt-components-sticky/sticky.twig" %}
        {% block sticky_content %}
          <div style="color:white;background-color:red">
            First sticky item
          </div>
        {% endblock %}
      {% endembed %}

      <div style="height:100vh;background-color:lightgray;"></div>

      {% embed "@bolt-components-sticky/sticky.twig" %}
        {% block sticky_content %}
          <div style="background-color: white">
            Second sticky item
          </div>
        {% endblock %}
      {% endembed %}

      <div style="height:100vh;background-color:lightgray;"></div>

      {% embed "@bolt-components-sticky/sticky.twig" %}
        {% block sticky_content %}
          <div style="color:white;background-color:blue">
            Third sticky item
          </div>
        {% endblock %}
      {% endembed %}

      <div style="height:100vh;background-color:lightgray;"></div>
        `,
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
