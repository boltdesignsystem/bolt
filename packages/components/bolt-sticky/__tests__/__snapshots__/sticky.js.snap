// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-sticky> Component basic usage 1`] = `
<div style="height:50vh;background-color:lightgray;">
</div>
<bolt-sticky bolt-component
             class="c-bolt-sticky js-bolt-sticky"
>
  <div style="color:white;background-color:red">
    First sticky item
  </div>
</bolt-sticky>
<div style="height:100vh;background-color:lightgray;">
</div>
<bolt-sticky bolt-component
             class="c-bolt-sticky js-bolt-sticky"
>
  <div style="background-color: white">
    Second sticky item
  </div>
</bolt-sticky>
<div style="height:100vh;background-color:lightgray;">
</div>
<bolt-sticky bolt-component
             class="c-bolt-sticky js-bolt-sticky"
>
  <div style="color:white;background-color:blue">
    Third sticky item
  </div>
</bolt-sticky>
<div style="height:100vh;background-color:lightgray;">
</div>
`;
